### PR TITLE
Dashing migration

### PR DESCRIPTION
**Pull request by julionce**
This pull request adapts these demos to Dashing release.
These demos has been reorganized in different folder `c -> rclc` and `cpp -> rclcpp`.
`rcl` porting has not been done yet.

### CODE REVIEW

File: rcl/int32_publisher/package.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>int32_publisher_rcl</name>
+  <version>0.0.1</version>
+  <description>Example of a publisher using int32</description>
+  <maintainer email="julianbermudez@eprosima.com">Julian Bermudez</maintainer>
+  <license>Apache License 2.0</license>
```
Comment (ralph-lange) [inline rcl/int32_publisher/package.xml]:
```suggestion
  <license>Apache-2.0</license>
```
Cf. previous comment and https://spdx.org/licenses/

File: rcl/int32_subscriber/main.c [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    for (size_t i = 0; i < wait_set.size_of_subscriptions; ++i) {
+      rv = rcl_take(wait_set.subscriptions[i], msg, NULL, NULL);
+      if (RCL_RET_OK == rv)
+      {
+        printf("I received: [%i]\n", ((const std_msgs__msg__Int32*)msg)->data);
+      }
+    }
+  } while (rv);
+
+  rv = rcl_subscription_fini(&subscription, &node);
```
Comment (ralph-lange) [inline rcl/int32_subscriber/main.c]:
```suggestion
  rcl_subscription_fini(&subscription, &node);
```
rv is not used.

File: rcl/int32_subscriber/main.c [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  rv = rcl_node_fini(&node);
```
Comment (ralph-lange) [inline rcl/int32_subscriber/main.c]:
```suggestion
  rcl_node_fini(&node);
```
rv is not used.

File: rclcpp/complex_msg_publisher/main.cpp [hunk 4]
```diff
@@ -27,8 +27,9 @@ class complex_msg_publisher_cpp_node : public rclcpp::Node
   complex_msg_publisher_cpp_node()
   : Node("complex_msg_publisher_cpp")
   {
+    rclcpp::QoS qos(10);
     publisher_ = this->create_publisher<complex_msgs::msg::NestedMsgTest>(
-      "complex_msgs_msg_NestedMsgTest");
+      "complex_msgs_msg_NestedMsgTest", qos);
```
Comment (ralph-lange) [inline rclcpp/complex_msg_publisher/main.cpp]: You may want to use rclcpp::SystemDefaultsQoS() instead.

File: rclcpp/complex_msg_subscriber/main.cpp [hunk 5]
```diff
@@ -26,8 +26,9 @@ class complex_msg_subscriber_cpp_node : public rclcpp::Node
   complex_msg_subscriber_cpp_node()
   : Node("complex_msg_subscriber_cpp")
   {
+    rclcpp::QoS qos(10);
     subscription_ = this->create_subscription<complex_msgs::msg::NestedMsgTest>(
-      "complex_msgs_msg_NestedMsgTest",
+      "complex_msgs_msg_NestedMsgTest", qos,
```
Comment (ralph-lange) [inline rclcpp/complex_msg_subscriber/main.cpp]: Possible to use rclcpp::SystemDefaultsQoS() instead?

**Review by ralph-lange — COMMENTED** [review]
Looks good to me. I like the proper renaming of the packages with suffixes _rclcpp or _rclc. In the next months, we should get rid of rclc completely.

File: rcl/int32_publisher/package.xml (hunk 1 above)
Comment (julibert) [reply rcl/int32_publisher/package.xml]: Thanks for the information. Change!

File: rcl/int32_subscriber/main.c (hunk 2 above)
Comment (julibert) [reply rcl/int32_subscriber/main.c]: It is outdated, new we use macros for return-value-checking.

File: rcl/int32_subscriber/main.c (hunk 3 above)
Comment (julibert) [reply rcl/int32_subscriber/main.c]: It is outdated, new we use macros for return-value-checking.

File: rclcpp/complex_msg_subscriber/main.cpp (hunk 5 above)
Comment (julibert) [reply rclcpp/complex_msg_subscriber/main.cpp]: Changed!

File: rclcpp/complex_msg_publisher/main.cpp (hunk 4 above)
Comment (julibert) [reply rclcpp/complex_msg_publisher/main.cpp]: Change!

**Pull request merged**